src/lava/magma/runtime/runtime.py

```diff
@@ -229,25 +229,19 @@ def _run(self, run_condition):
                         data = recv_port.recv()
                         if not enum_equal(data, MGMT_RESPONSE.DONE):
                             if enum_equal(data, MGMT_RESPONSE.ERROR):
-                                # Receive all error messages from service
-                                error_msgs = []
-                                for p in self.service_to_runtime_data:
-                                    if p.probe():
-                                        num_errors = int(p.recv()[0])
-                                        for i in range(num_errors):
-                                            num_bytes = int(p.recv()[0])
-                                            data = []
-                                            for i in range(num_bytes):
-                                                data.append(int(p.recv()[0]))
-                                            error_msgs.append(
-                                                bytes(data).decode())
-
-                                # Gather all exceptions
-                                exceptions = []
-                                for msg in error_msgs:
-                                    exceptions.append(Exception(msg))
-
-                                raise Exception(exceptions)
+                                # Receive all errors from the ProcessModels
+                                error_cnt = 0
+                                for actors in \
+                                        self._messaging_infrastructure.actors:
+                                    actors.join()
+                                    if actors.exception:
+                                        _, traceback = actors.exception
+                                        print(traceback)
```

Comment:
Does this actually identify the problem from which the error came? Perhaps attach this to the Exception object.
You should print the class name, Process name and id.

Reply:
To be clear. Both the processes should identify themselves but also the RuntimeServices in case the error happened in the RuntimeService.
Perhaps you even want to distinguish them clearly in the command line prints.

Perhaps you even want to first print a summary of everything that has thrown any errors at the top of the command line to orient the user because otherwise there could be quite a messy stack trace.

Reply:
The printed traceback leads you to the line of code the exception occurred, e.g., "line 50, in run_spk
    raise AssertionError("All the error info")" -> line 50 is within PyProcModel1. So you know exactly which exception happened and on which code line.

These tracebacks are printed for every exception occurred in all ProcessModels.